File: app/src/main/java/wangdaye/com/geometricweather/basic/model/weather/History.java
package wangdaye.com.geometricweather.basic.model.weather;

import java.util.Date;

/**
 * History.
 *
 * All properties are {@link androidx.annotation.NonNull}.
 * */
public class History {

    private Date date;
    private long time;

    private int daytimeTemperature;
    private int nighttimeTemperature;

    public History(Date date, long time, int daytimeTemperature, int nighttimeTemperature) {
        this.date = date;
        this.time = time;
        this.daytimeTemperature = daytimeTemperature;
        this.nighttimeTemperature = nighttimeTemperature;
    }

    public Date getDate() {
        return date;
    }

    public long getTime() {
        return time;
    }

    public int getDaytimeTemperature() {
        return daytimeTemperature;
    }

    public int getNighttimeTemperature() {
        return nighttimeTemperature;
    }
}

File: app/src/main/java/org/breezyweather/common/basic/models/weather/History.kt
package org.breezyweather.common.basic.models.weather

import java.io.Serializable
import java.util.Date

/**
 * History.
 */
class History(
    val date: Date,
    val daytimeTemperature: Int? = null,
    val nighttimeTemperature: Int? = null
) : Serializable
